File: a693(dynasty_wars)/Python/dynasty_wars.py
while True:
    try:
        n, m = map(int, input().split())
        satiety = list(map(int, input().split()))
        partial_sum_of_satiety = [0]
        cases = []
        sum = 0
        for i in range(n):
            sum += satiety[i]
            partial_sum_of_satiety.append(sum)
        for i in range(m):
            begin, end = map(int, input().split())
            cases.append(partial_sum_of_satiety[end] - partial_sum_of_satiety[begin - 1])
        for case in cases:
            print(case)
    except EOFError:
        break

File: a693(dynasty_wars)/C++/dynasty_wars.cpp
#include <iostream>

int main(int argc, char const *argv[]) {
  std::ios_base::sync_with_stdio(false);
  std::cin.tie(0);
  int n, m;
  while (std::cin >> n >> m) {
    int satiety[n];
    int partial_sum_of_satiety[n + 1];
    int sum = 0;
    partial_sum_of_satiety[0] = 0;
    for (int i = 0; i < n; i++) {
      std::cin >> satiety[i];
      sum += satiety[i];
      partial_sum_of_satiety[i + 1] = sum;
    }
    for (int i = 0; i < m; i++) {
      int begin, end;
      std::cin >> begin >> end;
      std::cout << partial_sum_of_satiety[end] - partial_sum_of_satiety[begin - 1] << '\n';
    }
  }
  return 0;
}
